pkg/controller/release/release_controller.go

```diff
@@ -350,11 +350,16 @@ func (c *Controller) syncOneReleaseHandler(key string) error {
 			err.Error(),
 		)
 		releaseutil.SetReleaseCondition(&rel.Status, *condition)
+	}
 
-		if _, updateErr := c.clientset.ShipperV1alpha1().Releases(namespace).Update(rel); updateErr != nil {
-			return shippererrors.NewKubeclientUpdateError(rel, updateErr)
+	// any better ideas?
+	if !reflect.DeepEqual(curRel, rel) {
+		if _, err := c.clientset.ShipperV1alpha1().Releases(namespace).Update(rel); err != nil {
+			return shippererrors.NewKubeclientUpdateError(rel, err)
 		}
+	}
 
+	if err != nil {
```

Comment:
What is this guarding against?

Reply:
Renamed this err to scheduleErr. The idea is to get control back from ScheduleRelease routine, check errors and if there are some, update corresponding conditions; then check if there are any effective changes to the release after all these cycles. Afterwards it checks if there was an error from ScheduleRelease and if there was, exit the routine.